sdk/core/System.ClientModel/src/Message/PipelineResponse.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Buffers;
+using System.ClientModel.Internal;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+public abstract class PipelineResponse : IDisposable
+{
+    // TODO: We can move this onto ClientPipeline once we move that type into main.
+    internal static TimeSpan DefaultNetworkTimeout { get; } = TimeSpan.FromSeconds(100);
```

Comment:
Just calling out that this more rightly lives on ClientPipeline, probably, but since that type isn't coming in through this PR, I'm stashing this property here for now to allow a sensible default timeout for the new response buffering logic that we've moved onto PipelineResponse.